R/jhu_data.R

```diff
@@ -68,102 +73,176 @@ load_jhu_data <- function(
       dplyr::summarize(cum = sum(cum)) %>%
       dplyr::group_by(location_name) %>%
-      dplyr::mutate(inc = cum - dplyr::lag(cum, 1L)) %>%
+      dplyr::mutate(inc = diff(c(0, cum))) %>%
       dplyr::ungroup() %>%
       dplyr::left_join(
         covidData::fips_codes %>% dplyr::filter(nchar(location) == 2),
-        by = 'location_name') %>%
+        by = "location_name"
+      ) %>%
       dplyr::select(location, date, cum, inc)
-    
+
     results <- dplyr::bind_rows(results, state_results)
   }
 
   # summarized results for national level
-  if('national' %in% spatial_resolution) {
-    # because we don't filter on states_to_keep as above, we are off by a total
+  if ("national" %in% spatial_resolution) {
+    # because we don"t filter on states_to_keep as above, we are off by a total
     # of 3 deaths attributed to Diamond Princess.
     national_results <- jhu_data %>%
       dplyr::group_by(date) %>%
       dplyr::summarize(cum = sum(cum)) %>%
       dplyr::ungroup() %>%
       dplyr::mutate(
-        inc = cum - dplyr::lag(cum, 1L),
-        location = 'US'
+        inc = diff(c(0, cum)),
+        location = "US"
       ) %>%
       dplyr::select(location, date, cum, inc)
 
     results <- dplyr::bind_rows(results, national_results)
   }
+
   
+   if (replace_negatives == TRUE) {
+    results = covidData::replace_negatives(data = results, measure = measure)
+   }
+
   # TODO: in results data frame, replace daily inc with NA in specific rows, if requested
   # at this point, the results data frame will have daily incidence values and we want to
   # replace the numbers in some rows with NAs (no new rows, editing existing rows)
-  
-  # TODO: if temporal_resolution == 'weekly', aggregate daily incidence to weekly incidence here
+
+  if (adjustment_cases != "none" & length(adjustment_cases) > 0) {
+    adjustment_states <- sub("-.*", "", adjustment_cases)
+    adjustment_dates <- sub("^.*?-", "", adjustment_cases)
+    adjustment_state_fips <- unlist(lapply(
+      adjustment_states, function(x) {
+        covidData::fips_codes[which(covidData::fips_codes$abbreviation == x), ]$location
+      }
+    ))
+    
+
+    adjustments <- data.frame(fips = adjustment_state_fips, dates = as.Date(adjustment_dates))
+    if (adjustment_method == "fill_na") {
+      results <- covidData::fill_na(results = results, adjustments = adjustments)
+    }
+
+    
+    if (adjustment_method == "impute_and_redistribute") {
+       if (replace_negatives == FALSE) {
+         results = covidData::replace_negatives(data = results, measure = measure)
+       }
+
+      for (i in 1:nrow(adjustments)) {
+        adjustment_location <- adjustments[i, ]$fips
+        adjustment_date <- as.Date(adjustments[i, ]$date)
+
+        # Get state, counties and national observations for an adjustment case
+        location_data <- results %>%
+          dplyr::filter(stringr::str_sub(location, start = 1, end = 2) %in% adjustment_location |
```

Comment:
Using comparison with `==` rather than `%in%` might be a little clearer.

```suggestion
          dplyr::filter(stringr::str_sub(location, start = 1, end = 2) == adjustment_location |
```

Reply:
This is for getting state-level data when adjustment_location is a county, and also the other way around. So it's only matching the first two digits (state part) of a fips code.